asllib/types.ml

```diff
@@ -111,47 +106,44 @@ let is_builtin_singular ty =
 let is_builtin_aggregate ty =
   match ty.desc with
   | T_Tuple _ | T_Array _ | T_Record _ | T_Exception _ -> true
-  | _ -> false
+  | _ -> false |: TypingRule.BuiltinAggregateType 
 
-let is_builtin ty = is_builtin_singular ty || is_builtin_aggregate ty
-let is_named ty = match ty.desc with T_Named _ -> true | _ -> false
+let is_builtin ty = is_builtin_singular ty || is_builtin_aggregate ty |: TypingRule.BuiltinSingularOrAggregate
```

Comment:
Do we really need to name this with explicit annotations? I doubt that anybody will use this to test anything and this could probably be a comment?